src/test/java/de/qabel/desktop/hockeyapp/HockeyAppTest.java

```diff
@@ -0,0 +1,167 @@
+package de.qabel.desktop.hockeyapp;
+
+import de.qabel.core.accounting.CloseableHttpClientStub;
+import de.qabel.core.accounting.CloseableHttpResponseStub;
+import org.apache.http.NameValuePair;
+import org.apache.http.entity.BasicHttpEntity;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class HockeyAppTest {
+
+    private final String shortVersion = "1.1";
+
+    private CloseableHttpClientStub httpClient = new CloseableHttpClientStub();
+
+    private HockeyAppConfiguration config = new HockeyAppConfiguration(shortVersion, httpClient);
+    private VersionClient versionClient = new VersionClient(config, httpClient);
+    private HockeyApp hockeyApp = new HockeyApp(shortVersion, httpClient);
+
+    private String feedback;
+    private String name;
+    private String email;
+    private String formattedStacktrace;
+
+    public HockeyAppTest() {
+        versionClient.setVersion(new HockeyAppVersion(1, config.getAppVersion()));
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        feedback = "HockeyAppTest feedback text";
+        name = "HockeyAppTest";
+        email = "HockeyAppTest@desktop.de";
+        formattedStacktrace = "XCEPTION REASON STRING\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "ANOTHER EXCEPTION REASON STRING\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)";
+
+        stubFeedbackResponse();
+        stubVersionToApp();
+    }
+
+
+    @Test
+    public void testSendFeedback() throws IOException {
+
+        hockeyApp.sendFeedback(feedback, name, email);
+        String responseContent = getStubFeedbackResponseContent();
+        JSONObject json = parseFeedbackJson(responseContent);
+
+        assertEquals(123, json.get("id"));
+        assertEquals(shortVersion, versionClient.getVersion().getShortVersion());
```

Comment:
still: the intent of this test is very unclear. You send feedback so I would expect that you check the request content to verify that the content that was sent is corrent. But you check the responseContent for an id "123" .. what's that magic number and why should anyone care about it?